cmd/create.go

```diff
@@ -66,23 +68,11 @@ var createCmd = &cobra.Command{
 			err = fmt.Errorf("directory '%s' already exists and is not empty. Please choose a different name", flagName)
 			cobra.CheckErr(textinput.CreateErrorInputModel(err).Err())
 		}
-		flagFramework := cmd.Flag("framework").Value.String()
-		flagDBDriver := cmd.Flag("driver").Value.String()
 
-		if flagFramework != "" {
-			isValid := isValidProjectType(flagFramework, allowedProjectTypes)
-			if !isValid {
-				err = fmt.Errorf("project type '%s' is not valid. Valid types are: %s", flagFramework, strings.Join(allowedProjectTypes, ", "))
-				cobra.CheckErr(textinput.CreateErrorInputModel(err).Err())
-			}
-		}
-
-		if flagDBDriver != "" {
-			isValid := isValidDBDriver(flagDBDriver, allowedDBDrivers)
-			if !isValid {
-				cobra.CheckErr(fmt.Errorf("database driver '%s' is not valid. Valid types are: %s", flagDBDriver, strings.Join(allowedDBDrivers, ", ")))
-			}
-		}
+		// VarP already validates the contents of the framework flag.
+		// If this flag is filled, it is always valid
+		flagFramework := flags.Framework(cmd.Flag("framework").Value.String())
+		flagDBDriver := flags.Database(cmd.Flag("driver").Value.String())
```

Comment:
How does error handling on this work then?

Reply:
The validations happen by `VarP` that function registers the flags and only allows for valid values to pass through. If there is an invalid value, `VarP` prints the last arg to the terminal and exists the program. An empty string is passed if the flag is not used on the command line. I think technically you can also pass an empty string (`go-blueprint -f "") but I don't think we should prevent this since it is already handled the same as not using the flag at all.


https://github.com/MitchellBerend/go-blueprint/blob/e443e0d9a346b69448f1d46419c0048418998dd0/cmd/create.go#L45-L46